tests/components/spotify/test_config_flow.py

```diff
@@ -0,0 +1,139 @@
+"""Tests for the Spotify config flow."""
+from unittest.mock import patch
+
+from spotipy import SpotifyException
+
+from homeassistant import data_entry_flow, setup
+from homeassistant.components.spotify.const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_USER, SOURCE_ZEROCONF
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from tests.common import MockConfigEntry
+
+
+async def test_abort_if_no_configuration(hass):
+    """Check flow aborts when no configuration is present."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "missing_configuration"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_ZEROCONF}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "missing_configuration"
+
+
+async def test_zeroconf_abort_if_existing_entry(hass):
+    """Check zeroconf flow aborts when an entry already exist."""
+    MockConfigEntry(domain=DOMAIN).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_ZEROCONF}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_full_flow(hass, aiohttp_client, aioclient_mock):
+    """Check a full flow."""
+    assert await setup.async_setup_component(
+        hass,
+        DOMAIN,
+        {
+            DOMAIN: {CONF_CLIENT_ID: "client", CONF_CLIENT_SECRET: "secret"},
+            "http": {"base_url": "https://example.com"},
+        },
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    # pylint: disable=protected-access
+    state = config_entry_oauth2_flow._encode_jwt(hass, {"flow_id": result["flow_id"]})
```

Comment:
Can we patch this encode function instead so we don't need to disable the pylint warning?

Reply:
I guess that will be unneeded hard since it has to be valid later on.
I followed the implementation by Paulus in Somfy for this.

Reply:
Ok. We should keep the disable so that editor linters and lazytox don't complain.

Reply:
In my opinion we should lint tests too.

Reply:
Totally agree! I lint my tests, but it has never been done @ HA. The amount of linter warnings is insane right now.